### ISSUE TITLE
Webpack init doesn't check for npm linked scaffolds

### ISSUE THREAD

**Issue by romerocs** [webpack/webpack-cli#1001]
I created a [basic scaffold](https://github.com/romerocs/webpack-scaffold-basic) using webpack's [documentation](https://webpack.js.org/guides/scaffolding/). I wanted to use the `npm link` so that I can set up my scaffold as a global module without publishing to npm.

### How to reproduce the problem:

Clone my [scaffold](https://github.com/romerocs/webpack-scaffold-basic) to a local machine.

Run `npm install` in the cloned package.  
Then run the `npm link` command.

 
This is the output I see after running the command:
`/usr/local/lib/node_modules/webpack-scaffold-basic -> /Users/firstname.lastname/Sandbox/webpack-scaffold-basic`

Next go to project that you want to scaffold and run the following command (per the documentation)
`webpack-cli init webpack-scaffold-basic`

I then get the following error message:  
`TypeError: Cannot resolve location of package webpack-scaffold-basic.`

Now according to the documentation for [npm link](https://docs.npmjs.com/cli/link), there's actually an extra step required to make this work. After running the `npm link` command from the scaffold, you then need to run the command `npm link package-name` from the folder where you want to generate the scaffold. So, I tried that and got the same `TypeError. Cannot resolve...` message.

I then went into the @webpack-cli/utils source code. I found the [npm-package-exists](https://github.com/webpack/webpack-cli/blob/next/packages/utils/npm-packages-exists.ts) utility. From what I can tell, it looks like the function npmPackageExists has only 2 options for specifying a scaffold, either as an absolute local path OR as a fully published npm package. My suggestion would be to add another method to check if the given scaffold is in fact an npm linked package.

I created an extremely not-ready-for-production method that checks the scaffold name against linked packages. I was able to get the scaffold to work using my linked scaffold name with this method.

```javascript
function getLinkedPath(pkg) {
  return new Promise(function(resolve, reject) {
    require("child_process").exec(
      `npm ls -g --depth 0 --link ${pkg}`,
      (err, stdout, stderr) => {
        if (err) return cb(err);
        const arr = stdout.split(" ");
        const path = arr[4].replace(/(\r\n|\n|\r)/gm, "");

        if (err) {
          reject(err);
        } else {
          resolve(path);
        }
      }
    );
  });
}
```

OS: macOS Mojave 10.14.5   

My globally installed package versions:
v11.6.0   
npm@6.10.0  
webpack@4.35.3  
webpack-cli@3.3.6  
@webpack-cli/generators@0.1.8  
@webpack-cli/init@0.2.2  
@webpack-cli/utils@0.2.2

**Comment by rishabh3112** [comment]
Thanks @romerocs for reporting.
I think it would be great addition, will look into the issue !

**Comment by rishabh3112** [comment]
@evenstensberg should I resolve this issue for v3 or directly v4 (as it would be our next major version)?

**Comment by anikethsaha** [comment]
Nice!  I am +1 for this addition
Just an addition
You can give the absolute path to your linked package too. 
Like this
```bash
webpack-cli init ./node_modules/[your linked scaffold name]
```

**Comment by ematipico** [comment]
V4

**Comment by evenstensberg** [comment]
If someone wants, submit a PR towards the next branch

**Comment by rishabh3112** [comment]
Fix has been sent to #1006. @ematipico have a look when free.
P.S. Tests are failing because whole test suite is broke and I am not allowed to fix 'em.
